mu4e/mu4e-context.el

```diff
@@ -72,13 +72,135 @@ none."
   that takes a message plist for the message replied to or
   forwarded, and nil otherwise. Before composing a new message,
   `mu4e' switches to the first context for which `match-func'
-  returns t."
+  returns t.
+- `vars': variables to set when entering context.
+- `user-mail-address': Defaults to the global value when the context is created.
+- `smtpmail-smtp-user': Defaults to the global value if non-nil when the context
+  is created, or the context `user-mail-address' otherwise.
+- `maildir': Mailbox folder name in as stored in `mu4e-maildir' (just the name,
+  there must be no '/').  Defaults to `name'.
+- `drafts-folder': Context value of `mu4e-drafts-folder'.  Defaults to
+  \"drafts\".
+- `sent-folder': Context value of `mu4e-sent-folder'.  Defaults to \"sent\".
+- `trash-folder': Context value of `mu4e-trash-folder'.  Defaults to \"trash\".
+- `refile-folder': Context value of `mu4e-refile-folder'.  Defaults to
+  \"refile\".
+- `no-trash-flag': If non-nil, the maildir will be added to
+  `mu4e-move-to-trash-patterns' so that trashing moves the message instead of flagging.
+- `predicate': A function that takes a message and returns non-nil if it matches
+  the context.  This is only used if `match-func' is not provided, in which case
+  the context is always matched against the message folder."
   name                      ;; name of the context, e.g. "work"
   (enter-func nil)          ;; function invoked when entering the context
   (leave-func nil)          ;; function invoked when leaving the context
   (match-func nil)          ;; function that takes a msg-proplist, and return t
   ;; if it matches, nil otherwise
-  vars)                     ;; alist of variables.
+  vars                      ;; alist of variables.
+  ;; We set sane defaults for the following variables.  They will be added to
+  ;; the context vars.
+  (user-mail-address user-mail-address)
+  (smtpmail-smtp-user smtpmail-smtp-user)
+  ;; Folders:
+  maildir
+  (drafts-folder "drafts")
+  (sent-folder "sent")
+  (trash-folder "trash")
+  (refile-folder "archive")
+  ;; Trash fix.
+  no-trash-flag
+  ;; Rule for matching the context.
+  predicate)
```

Comment:
`mu4e-context` is very simple and should stay unchanged, I think. So no extra variables etc. we have `vars` for that already; same for `predicate` and `match-func` already.

All the extra magic can go to `make-mu4e-context-account`.

Reply:
Updated: see #1390.